doc/developers/contributing.rst

```diff
@@ -603,6 +603,24 @@ Finally, follow the formatting rules below to make it consistently good:
     SelectKBest : Select features based on the k highest scores.
     SelectFpr : Select features based on a false positive rate test.
 
+* When documenting the parameters and attributes, have the following in mind:
+
+    1. Do not use optional. Use `default=`. `str {'a', 'b'} or float, default=1.0`
+    2. Python basic types. (`bool` instead of `boolean`)
+    3. When defining 1-D shape, use parenthesis. `array-like shape=(n_samples,), None, default=None`.
+    4. When defining 2-D shape, use parenthesis. `array-like shape=(n_samples, n_features)`.
```

Comment:
What about the case that the array can be both 1-D and 2-D (e.g. output `y_pred`)